ocaml/xapi/storage_migrate.ml

```diff
@@ -586,7 +610,11 @@ let stat ~dbg ~id =
     | (Some receive_state, _, _) ->
       receive_state.Receive_state.remote_vdi, receive_state.Receive_state.leaf_vdi
     | (_, Some send_state, _) ->
-      snd (of_mirror_id id), send_state.Send_state.mirror_vdi
+      let dst_vdi = match send_state.Send_state.remote_info with
+        | Some remote_info -> remote_info.Send_state.vdi
+        | None -> ""
```

Comment:
This doesn't look right, but I think that's more a problem with SMAPIv2 than your code.